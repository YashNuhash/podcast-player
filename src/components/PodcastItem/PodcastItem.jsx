import React from 'react'
import { Link } from 'react-router-dom'
import EllipsisText from '../EllipsisText/EllipsisText'
import styles from './PodcastItem.module.scss'

export const PodcastItem = ({ id, thumbnail, title, publisher }) => (
  <div className={styles.Item} key={id}>
    <div className={styles.ImageWrapper}>
      <img className={styles.Image} src={thumbnail} alt='' />
    </div>
    <div className={styles.Content}>
      <EllipsisText tagName='p' className={styles.Publisher}>
        {publisher}
      </EllipsisText>
      <EllipsisText tagName='h4' className={styles.Title}>
        {title}
      </EllipsisText>

      <Link to={`/podcast/${id}`} className={styles.Button}>
        View More
      </Link>
    </div>
  </div>
)

export default PodcastItem
