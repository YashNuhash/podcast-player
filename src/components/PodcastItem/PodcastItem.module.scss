@import '../../utils';

.Item {
    display: flex;
}

.Content {
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
}

.ImageWrapper {
    height: 15rem;
    width: 15rem;
    background-color: $bg;
    border-radius: 3rem;
    overflow: hidden;
    margin-right: 3rem;
    position: relative;
    flex-shrink: 0;
}

.Image {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
}

.Title,
.Publisher {
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 2;
    text-overflow: ellipsis;
    white-space: normal;
    overflow: hidden;
}

.Publisher {
    color: $light-text;
    font-weight: 600;
    font-size: 1.4rem;
}

.Button {
    padding: 0.5rem 1rem;
    text-align: center;
    border: 1px solid $primary;
    border-radius: 100px;
    color: $primary;
    text-decoration: none;
    align-self: flex-start;
    font-size: 1.5rem;
}

.Slider {
    flex-direction: column;

    .Image {
        margin-right: 0;
        margin-bottom: 1rem;
    }

    .Publisher {
        font-size: 1.3rem;
    }

    .Title {
        font-size: 1.5rem;
    }
}
