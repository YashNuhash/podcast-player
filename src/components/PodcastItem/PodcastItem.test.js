import React from 'react'
import { shallow } from 'enzyme'
import { PodcastItem } from './PodcastItem'

describe('PodcastItem', () => {
    it('should render', () => {
        const mockPodcast = {
            id: '1',
            thumbnail: 'http://www.example.com/link',
            publisher: 'John Doe',
            title: 'Podcast',
        }

        expect(shallow(<PodcastItem {...mockPodcast} />)).toMatchSnapshot()
    })
})
