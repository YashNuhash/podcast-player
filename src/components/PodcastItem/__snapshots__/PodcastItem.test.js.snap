// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PodcastItem should render 1`] = `
<div
  className="Item"
  key="1"
>
  <div
    className="ImageWrapper"
  >
    <img
      alt=""
      className="Image"
      src="http://www.example.com/link"
    />
  </div>
  <div
    className="Content"
  >
    <EllipsisText
      className="Publisher"
      tagName="p"
    >
      John Doe
    </EllipsisText>
    <EllipsisText
      className="Title"
      tagName="h4"
    >
      Podcast
    </EllipsisText>
    <Link
      className="Button"
      to="/podcast/1"
    >
      View More
    </Link>
  </div>
</div>
`;
